### PR TITLE
AlertState cache cleared on load

### PR DESCRIPTION
**Pull request by joonicks**
Loading a savegame would leave alertstate cache values undefined possibly triggering an alert or postponing further checks hundreds of years into the future.

This fixes it, clearing the cache on load and adding a sanity check in UpdateAlertState()

### CODE REVIEW

**Comment by fluffyfreak** [comment]
nice one!

File: src/Ship.cpp [hunk 1]
```diff
@@ -944,6 +954,16 @@ void Ship::UpdateAlertState()
 	}
 
 	bool ship_is_near = false, ship_is_firing = false;
+
+	// sanity check: m_lastAlertUpdate should not be in the future.
+	// reset and re-check if it is.
+	if (m_lastAlertUpdate > Pi::game->GetTime() + 2)
```
Comment (laarmen) [inline src/Ship.cpp]: Out of curiosity, where does the +2 come from?

File: src/Ship.cpp [hunk 2]
```diff
@@ -213,6 +218,8 @@ void Ship::InitMaterials()
 
 void Ship::Init()
 {
+	Output("Running Ship::Init()\n");
```
Comment (laarmen) [inline src/Ship.cpp]: I'm assuming this is a leftover debug printout? In any case, I'd rather not have it, as it clutters the output without adding much information.

File: src/Ship.cpp (hunk 1 above)
Comment (joonicks) [reply src/Ship.cpp]: arbitrary. could be +1 but. meh.

Comment (fluffyfreak) [reply src/Ship.cpp]: Shouldn't even need to plus anything since it's using `>`

**Comment by fluffyfreak** [comment]
Ok, deal with those two minor comments please and then I think it's good to merge!
Another good fix @joonicks :)

**Comment by impaktor** [comment]
For future reference, that last commit should ideally have been squashed with the previous.

At moments like these:
1. do the change
2. `git add` the change
3. `git commit --amend`
4. `git push -f` to your branch.

We try to have meaningful commit messages. "pimplesqueezer" doesn't say much...

But good that the requested changes were fixed fast, thanks.

**Pull request merged**